### PR TITLE
Completing information about discoverable

### PR DESCRIPTION
**Pull request by delapuente**
This is a feature, not a technology and it could be awesome if @wfwalker or somebody from product at @mozilla/wadi could review the information provided. It would be fantastic if we could provide some articles talking about _discoverability_.

### CODE REVIEW

**Comment by digitarald** [comment]
Another use case;

As a browser I want to give apps special capabilities and allow them to escape the tab.

**Comment by delapuente** [comment]
Perfect! To complete these kind of PR I'm going to give 24 hours each, then merge because some info is better that no info at all. But from almost always nothing at all is the best compared with bad grammar so I call @mykmelez as the grammar inquisitor if he pleases.

File: _features/discoverable.html [hunk 1]
```diff
@@ -2,34 +2,22 @@
 title: Discoverable
 slug: discoverable
 description: Are identifiable as applications and search engines can find them.
+usecases:
+- <p>Progressive Web Applications offer unique capabilities; as a developer, I want
```
Comment (mykmelez) [inline _features/discoverable.html]: Nit: I'd make this a full stop, as these sentences are quite distinct:

> Progressive Web Applications offer unique capabilities. As a developer,

File: _features/discoverable.html [hunk 2]
```diff
@@ -2,34 +2,22 @@
...
+  my web applications to be promoted in search engines when the user is using
```
Comment (mykmelez) [inline _features/discoverable.html]: Now that you've used the word "applications" once, I'd abbreviate it to "apps" here and below.

File: _features/discoverable.html [hunk 3]
```diff
@@ -2,34 +2,22 @@
...
+  modern platforms.</p>
```
Comment (mykmelez) [inline _features/discoverable.html]: Hmm, "modern platforms" isn't clear. I assume "platform" here means operating system platform—Windows, Mac, Linux—but it's actually the browser that needs to be "modern" enough for progressive apps to benefit from the progressive APIs.

Although I'm not sure even that is necessary in order for progressive apps to be promoted in search engines. That could happen even in legacy browsers, since it's a function of the search engine.

This minimal change here is to replace "modern platforms" with "modern browsers." But I'd go further and strip "when the user is using modern platforms," unless we think search engines won't highlight apps on older browsers.

File: _features/discoverable.html [hunk 4]
```diff
@@ -2,34 +2,22 @@
...
+- <p>As a search engine owner I want to highlight those applications with best UX
```
Comment (mykmelez) [inline _features/discoverable.html]: Nit: here and below, append a comma to the "as a … clause, i.e. `As a search engine owner,`.

Comment (mykmelez) [inline _features/discoverable.html]: Also, "with best UX" should have a comma after it:

> with best UX, promoting them to the upper positions."

But "best UX" is a bit misleading, since apps with an awful user experience can still publish web manifests. And "upper positions" is a bit oblique, even though I figured out what it meant (position in search results).

I'd reword this user story to something like, "As a search engine owner, I want to highlight apps that provide high-quality icons and other meta-data, promoting them in search results."

File: _features/discoverable.html [hunk 5]
```diff
@@ -2,34 +2,22 @@
...
+  promoting them to the upper positions.</p>
+- <p>As a browser developer I want to give apps special capabilities and allow
+  them to <em>escape the tab</em>.</p>
```
Comment (mykmelez) [inline _features/discoverable.html:11]: "Escape the tab" is a great phrase!

File: _features/discoverable.html [hunk 6]
```diff
@@ -2,34 +2,22 @@
...
+- <p>As a marketplace owner I want to gather information about progressive web
+  application to rating and sorting them.</p>
```
Comment (mykmelez) [inline _features/discoverable.html]: "Marketplace" is a bit ambiguous here. I would prefix it with "app," i.e. call it an "app marketplace." Note also that the [Wikipedia entry for the concept](https://en.wikipedia.org/wiki/App_store) uses the phrase "app store" as the primary name for the concept, although it does reference "app marketplace" as an alternative.

Also, nits:
- application -> applications (or, preferably, "apps")
- rating and sorting -> rate and sort (although I would probably use a word like "categorize" or "organize" here)

File: _features/discoverable.html [hunk 7]
```diff
@@ -2,34 +2,22 @@
...
-  <li><a href="https://github.com/mozilla/oghliner" target="_blank">oghliner</a> is not only a template but a tool for deploying Offline Web Apps to GitHub Pages.</li>
-  <li><a href="https://www.talater.com/upup/" target="_blank">UpUp</a>is a tiny script that makes sure your site is always there for your users.</li>
-</ul>
--->
+<p>As soon as you start writing progressive web applications you want the World to
+  know about them. If a user visits your application in a modern platform, all your
+  web site's progressive super-powers simply emerege but, <em>what if the user
+  never heard about you?</em> Progressive Web Applications are discoverable
+  because they allow search engines and Web indices to find your application and
+  highlight their capabilities.</p>
```
Comment (mykmelez) [inline _features/discoverable.html]: Nits:
- append comma to `As soon as you start writing progressive web applications`
- World -> world
- modern platform -> modern browser
- emerege -> emerge
- but, -> , but
- never heard about you -> has never heard about you
- are discoverable -> are more discoverable (is what I think you meant to say)
- find your application and highlight their capabilities -> find your application and highlight _its_ capabilities (although you use _applications_ plural earlier in the sentence and _application_ singular here, and it might be better to instead use the plural here too)
- Web indices -> Web indexes (but see below)

Erm, but I don't think web manifests actually _allow_ search engines to find your app. They just enrich the meta-data available to search engines once they've already found your app. I would say something like "Progressive Web Apps are more discoverable because they enhance the information that search engines and Web indexes use to categorize and highlight your app." (Unsure if that's quite right; might need to wordsmith it some more.)

About indexes vs. indices, both are right! And they've both been in use for hundreds of years. There are some regional differences—the US uses _indexes_ more, while the UK uses _indices_ more—but there's no clear "winner," so my nit in this case is really just an opinion/preference (and perhaps I am biased because I'm US-based).

However: _indexes_ employs a more common pluralization rule (append -s/-es), while _indices_ employs a Latin-derived rule that doesn't apply to many English words, and I tend to prefer more common constructions in technical writing, as I think they reduce the potential for confusion. Someone who speaks English as a second language, and doesn't know the word, is more likely to understand _indexes_ than _indices_ as the pluralization of _index_ (unless perhaps they have experience with Latin or a Romance language).

Note also [Indexes vs. indices](http://grammarist.com/usage/indexes-indices/), which does a good job comparing the two words, and [Grammar: Indexes vs. Indices](https://federalist.wordpress.com/2006/09/28/grammar-indexes-vs-indices/), which argues for the latter to avoid confusion with the verb _indexes_.

Comment (mykmelez) [inline _features/discoverable.html]: Erm: one last note! When you say "web indices," I think you're referring to sites like `Yahoo!`, in which case I'd instead use the phrase [web directory](https://en.wikipedia.org/wiki/Web_directory) to describe them.

**Comment by mykmelez** [comment]
> I call @mykmelez as the grammar inquisitor if he pleases.

Thanks, Inquisitor is a fantastic nickname for me! I very much do please. :wink:

File: _features/discoverable.html (hunk 7 above)
Comment (delapuente) [inline _features/discoverable.html]: > Erm, but I don't think web manifests actually allow search engines to find your app. They just enrich the meta-data available to search engines once they've already found your app.

Well, maybe it's a matter of wording but the search engine scraps the web and it finds documents. By inspecting these documents, it's able to see if there is a linked manifest. Once the search engine finds the manifest, it can consider the site is indeed an application and treat it in a singular way.

Comment (delapuente) [inline _features/discoverable.html]: > Note also Indexes vs. indices, which does a good job comparing the two words, and Grammar: Indexes vs. Indices, which argues for the latter to avoid confusion with the verb indexes.

I use _indices_ if there is a mathematical, scientific or formal context. I preferred to change it here because I don't want the tone of the text to be too formal.

**Comment by delapuente** [comment]
@mykmelez if you don't mind to take a second look? Discussions can be open as issues and we can continue improving the info in a follow-up. What do you think?

File: _features/discoverable.html (hunk 7 above)
Comment (mykmelez) [inline _features/discoverable.html]: > Well, maybe it's a matter of wording but the search engine scraps the web and it finds documents. By inspecting these documents, it's able to see if there is a linked manifest. Once the search engine finds the manifest, it can consider the site is indeed an application and treat it in a singular way.

In that case, I would simply change the word "find" to "identify," as the manifest allows the search engine to identify a site it has found as an app. That's also consistent with what you say earlier, when you describe "websites identifiable as applications."

File: _features/discoverable.html [hunk 8]
```diff
@@ -2,34 +2,21 @@
...
-<h2>Resources</h2>
-<ul class="resources">
-  <li><a href="https://github.com/GoogleChrome/sw-precache" target="_blank">sw-precache</a> is a node module to generate service worker code that will precache specific resources.</li>
-  <li><a href="https://github.com/mozilla/oghliner" target="_blank">oghliner</a> is not only a template but a tool for deploying Offline Web Apps to GitHub Pages.</li>
-  <li><a href="https://www.talater.com/upup/" target="_blank">UpUp</a>is a tiny script that makes sure your site is always there for your users.</li>
-</ul>
--->
+<p>As soon as you start writing progressive web apps, you want the world to
+  know about them. If a user visits your app in a modern browser, all your
+  web site's progressive super-powers simply emerge but, <em>what if the user
```
Comment (mykmelez) [inline _features/discoverable.html]: The comma is still out of place: `but,` -> `,but`

Also, "web site" is spelled as two words here, while elsewhere it's spelled as one "website." Either is fine, but the document should be consistent.

**Comment by delapuente** [comment]
Merging this.
